internal/server/data/encryption_key.go

```diff
@@ -1,32 +1,59 @@
 package data
 
 import (
+	"fmt"
 	mathrand "math/rand"
 
-	"gorm.io/gorm"
-
+	"github.com/infrahq/infra/internal/server/data/querybuilder"
 	"github.com/infrahq/infra/internal/server/models"
 )
 
-func CreateEncryptionKey(db GormTxn, key *models.EncryptionKey) (*models.EncryptionKey, error) {
+type encryptionKeysTable models.EncryptionKey
+
+func (e encryptionKeysTable) Table() string {
+	return "encryption_keys"
+}
+
+func (e encryptionKeysTable) Columns() []string {
+	return []string{"algorithm", "created_at", "deleted_at", "encrypted", "id", "key_id", "name", "root_key_id", "updated_at"}
+}
+
+func (e encryptionKeysTable) Values() []any {
+	return []any{e.Algorithm, e.CreatedAt, e.DeletedAt, e.Encrypted, e.ID, e.KeyID, e.Name, e.RootKeyID, e.UpdatedAt}
+}
+
+func (e *encryptionKeysTable) ScanFields() []any {
+	return []any{&e.Algorithm, &e.CreatedAt, &e.DeletedAt, &e.Encrypted, &e.ID, &e.KeyID, &e.Name, &e.RootKeyID, &e.UpdatedAt}
+}
+
+func CreateEncryptionKey(tx WriteTxn, key *models.EncryptionKey) error {
+	switch {
+	case key.Name == "":
+		return fmt.Errorf("a name is required")
```

Comment:
can we rely on this being logged in a way that it is clear the name being required refers to an encryption key?

Reply:
This error should be only a safeguard for us, not something ever seen by a user. It should never end up in logs.

It doesn't look like we wrap it in any more context. I'll add "EncryptionKey" to the message to make it more clear.